packages/next/src/server/dev/on-demand-entry-handler.ts

```diff
@@ -899,50 +907,46 @@ export function onDemandEntryHandler({
   }
 
   // Make sure that we won't have multiple invalidations ongoing concurrently.
-  const curEnsurePage = new Map<string, Promise<void | { err: Error }>>()
+  const active = new Map<string, Promise<void>>()
 
   return {
-    async ensurePage({
-      page,
-      clientOnly,
-      appPaths = null,
-      match,
-      isApp,
-    }: {
+    async ensurePage(params: {
       page: string
       clientOnly: boolean
       appPaths?: ReadonlyArray<string> | null
       match?: RouteMatch
       isApp?: boolean
     }) {
-      const ensureKey = JSON.stringify({ page, clientOnly, match, appPaths })
-      const pendingEnsure = curEnsurePage.get(ensureKey)
-
-      if (pendingEnsure) {
-        const res = await pendingEnsure
-
-        if (res && res.err) {
-          throw res.err
+      // The cache key here is composed of the elements that affect the
+      // compilation, namely, the page, whether it's client only, and whether
+      // it's an app page. This ensures that we don't have multiple compilations
+      // for the same page happening concurrently.
+      //
+      // We don't include the whole match because it contains match specific
+      // parameters (like route params) that would just bust this cache. Any
+      // details that would possibly bust the cache should be listed here.
+      const key = `${params.page}:${params.clientOnly}:${params.isApp}`
```

Comment:
This seems like its fixing a bug.

Is there a corresponding test you can add to prove it fixes the bug?

Reply:
The reason that I changed this is because the `match` that was previously used was both quite large and would bust the cache too early (as it contains route matches which isn't actually used by the entry handler that we're memoizing for).